### PR TITLE
Clinicaltrial & disease & treatment

### PR DESCRIPTION
**Pull request by EmiliaCXY**

### CODE REVIEW

File: src/main/java/org/cpsc304/bigdata/db/dao/ClinicalTrialTreatmentDAOImpl.java [hunk 1]
```diff
@@ -50,46 +56,217 @@ public void add(ClinicalTrial trial, Treatment treatment) {
...
+                                set.getString("TrialName"),
+                                set.getString("Type"),
+                                set.getInt("Num_Participants"),
+                                set.getInt("IsComplete")
+                                ));
+            }
+            return clinicalTrails;
+        } catch (SQLException e) {
+            logger.warn(e.getMessage());
+            return null;
```
Comment (chuck-sys) [inline src/main/java/org/cpsc304/bigdata/db/dao/ClinicalTrialTreatmentDAOImpl.java]:
```suggestion
            return Collections.emptyList();
```

File: src/main/java/org/cpsc304/bigdata/db/dao/ClinicalTrialTreatmentDAOImpl.java [hunk 2]
```diff
@@ -50,46 +56,217 @@ public void add(ClinicalTrial trial, Treatment treatment) {
...
+                                set.getString("TreatmentName"),
+                                set.getFloat("Efficiency"),
+                                set.getString("Equipment"),
+                                set.getString("Risks")
+                        ));
+            }
+            return treatments;
+        } catch (SQLException e) {
+            logger.warn(e.getMessage());
+            return null;
```
Comment (chuck-sys) [inline src/main/java/org/cpsc304/bigdata/db/dao/ClinicalTrialTreatmentDAOImpl.java]:
```suggestion
            return Collections.emptyList();
```

File: src/main/java/org/cpsc304/bigdata/db/dao/ClinicalTrialTreatmentDAOImpl.java [hunk 3]
```diff
@@ -50,46 +56,217 @@ public void add(ClinicalTrial trial, Treatment treatment) {
...
+                        new ClinicalTrial(set.getString("TrialName"),
+                                "",
+                                0,
+                                1
+                        ));
+            }
+            return clinicalTrials;
+        } catch (SQLException e) {
+            logger.warn(e.getMessage());
+            return null;
```
Comment (chuck-sys) [inline src/main/java/org/cpsc304/bigdata/db/dao/ClinicalTrialTreatmentDAOImpl.java]:
```suggestion
            return Collections.emptyList();
```

File: src/main/java/org/cpsc304/bigdata/db/dao/DiseaseDAOImpl.java [hunk 4]
```diff
@@ -77,6 +77,42 @@
         }
     }
 
+    @Override
+    public List<Disease> findDiseaseByPara(final String table, final String number) {
```
Comment (chuck-sys) [inline src/main/java/org/cpsc304/bigdata/db/dao/DiseaseDAOImpl.java:82]: What does `Para` mean?

File: src/main/java/org/cpsc304/bigdata/db/dao/DiseaseDAOImpl.java [hunk 5]
```diff
@@ -77,6 +77,42 @@
...
+                        set.getString("Name"),
+                        set.getInt("Prevalence"),
+                        set.getString("Symptoms")));
+            }
+            return diseases;
+
+        } catch (SQLException e) {
+            logger.warn(e.getMessage());
+        }
+        return null;
```
Comment (chuck-sys) [inline src/main/java/org/cpsc304/bigdata/db/dao/DiseaseDAOImpl.java]:
```suggestion
        return Collections.emptyList();
```

File: src/main/resources/static/js/query.js [hunk 6]
```diff
@@ -34,4 +56,18 @@ function deleteDisease(name) {
...
+
+function populateTreatment(data) {
+  $('#treatment-table tbody').remove();
+  let rows = "";
+  for(const row of data){
+    rows += `<tr><td>${row.treatmentName}</td>
+                 <td>${row.efficiency}</td>
+                 <td>${row.equipment}</td>
+                 <td>${row.risks}</td>
+                 <td><a class="btn btn-info" href="https://en.wikipedia.org/wiki/${row.treatmentName}"> More Info</a></td></tr>`;
```
Comment (chuck-sys) [inline src/main/resources/static/js/query.js:109]: Nice:)

**Review by chuck-sys — CHANGES_REQUESTED** [review]
Good cleanup on the query. Just a few things.

File: src/main/java/org/cpsc304/bigdata/db/dao/DiseaseDAOImpl.java (hunk 4 above)
Comment (EmiliaCXY) [reply src/main/java/org/cpsc304/bigdata/db/dao/DiseaseDAOImpl.java:82]: just mean that users can cross reference disease with either clinical trial or treatment and find the disease with more than X clinical trials or treatments

File: src/main/resources/static/js/query.js (hunk 6 above)
Comment (EmiliaCXY) [reply src/main/resources/static/js/query.js:109]: oh haha well will be fun when we demo it

File: src/main/resources/static/js/query.js [hunk 7]
```diff
@@ -15,6 +16,26 @@ $(document).ready(() => {
         $.post(`/disease/link/${dName}/${tName}`, alert);
     });
 
+<<<<<<<<< Temporary merge branch 1
```
Comment (chuck-sys) [inline src/main/resources/static/js/query.js]:
```suggestion
```

File: src/main/resources/static/js/query.js [hunk 8]
```diff
@@ -117,4 +153,5 @@ function deletePatient(id) {
             $('#select-patient').click();
         }
     });
+>>>>>>>>> Temporary merge branch 2
```
Comment (chuck-sys) [inline src/main/resources/static/js/query.js]:
```suggestion
```

File: src/main/resources/templates/query.html [hunk 9]
```diff
@@ -1,3 +1,501 @@
+<<<<<<<<< Temporary merge branch 1
```
Comment (chuck-sys) [inline src/main/resources/templates/query.html]:
```suggestion
```

File: src/main/resources/templates/query.html [hunk 10]
```diff
@@ -522,3 +1020,4 @@ <h2 id="medical_record">Medical Record</h2>
 
 
 </html>
+>>>>>>>>> Temporary merge branch 2
```
Comment (chuck-sys) [inline src/main/resources/templates/query.html]: ??

**Review by chuck-sys — CHANGES_REQUESTED** [review]
Artifacts that need removing from merge.

File: src/main/resources/templates/query.html (hunk 10 above)
Comment (EmiliaCXY) [reply src/main/resources/templates/query.html]: merging issue? I guess IntelliJ did that

**Review by chuck-sys — APPROVED** [review]

**Pull request merged**